core/sr-api-macros/src/impl_runtime_apis.rs

```diff
@@ -471,13 +473,25 @@ impl<'a> Fold for ApiRuntimeImplToApiRuntimeApiImpl<'a> {
 
 			let runtime_mod_path = self.runtime_mod_path;
 			let runtime = self.runtime_type;
-			let arg_names2 = arg_names.clone();
-			let fn_name = prefix_function_with_trait(self.impl_trait_ident, &input.sig.ident);
-			let native_call_generator_ident =
-				generate_native_call_generator_fn_name(&input.sig.ident);
+			let mut arg_names2 = arg_names.clone();
+			let mut fn_name = prefix_function_with_trait(self.impl_trait_ident, &input.sig.ident);
 			let trait_generic_arguments = self.trait_generic_arguments;
 			let node_block = self.node_block;
 
+			let mut execution_context = parse_quote!( ExecutionContext::Other );
+			let mut call_generator_name = input.sig.ident.to_string();
+			
+			let with_context_str = "_with_context";
```

Comment:
Instead of doing all this string parsing here, I would propose to drop `generate_impl_with_context()` and these modifications. 

Switch to use `fold_impl_item`. Iterate manual over all methods and call the function that generates the method once with the original `impl_item_method` and a second time where you just replace the name and add the corresponding function argument.

Reply:
Yeah, makes much more sense :+1: .